src/dirtyfields/dirtyfields.py

```diff
@@ -79,6 +80,11 @@ def _as_dict(self, check_relationship, include_primary_key=True):
 
             field_value = getattr(self, field.attname)
 
+            if isinstance(field_value, File):
+                # Uses the path for files due to a perfomance regression caused by Django 3.1.
+                # For more info see: https://github.com/romgar/django-dirtyfields/issues/165
+                field_value = field_value.path if field_value else ''
```

Comment:
I think there are a couple of problems with using the `field_value.path` attribute

1. strictly speaking the `django.core.files.File` class (which you're checking for) isn't guaranteed to have the `path` attribute.
2. If `field_value` is a `FieldFile` it does have a `path` attribute, but that attribute is the result of `Storage.path()` which is likely to raise `NotImplementedError` for projects using remote file storage backends (e.g. S3, etc)

https://docs.djangoproject.com/en/4.0/ref/models/fields/#django.db.models.fields.files.FieldFile.path
https://docs.djangoproject.com/en/4.0/ref/files/storage/#django.core.files.storage.Storage.path

Maybe using the `name` attribute is a better choice
https://docs.djangoproject.com/en/4.0/ref/models/fields/#django.db.models.fields.files.FieldFile.name
https://docs.djangoproject.com/en/4.0/ref/files/file/#django.core.files.File.name

Reply:
Sure, no problem! This is why I also have the guard. I actually thought about the `name` but was not sure if it was ok to use based on the ticket. Will try to fix soon. Thanks for the review :+1: